pytorch_lightning/profiler/profilers.py

```diff
@@ -466,15 +466,22 @@ def _create_profiler(self, action_name, profiler, enter=True):
         profiler_args = {k: v for k, v in vars(self).items() if k in init_args}
         pr = profiler(**profiler_args)
         if enter:
-            pr = pr.__enter__()
+            out_pr = pr.__enter__()
+            if out_pr is not None:
+                pr = out_pr
         self.profiler = pr
+        return pr
```

Comment:
```suggestion
        return self.profiler
```
why do you need this return?